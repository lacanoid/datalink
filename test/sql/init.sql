CREATE EXTENSION ddl;

\dx ddl

\dx+ ddl

