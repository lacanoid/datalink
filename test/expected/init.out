create extension datalink cascade;
NOTICE:  installing required extension "uuid-ossp"
NOTICE:  installing required extension "plperlu"
set search_path=datalink;
\dx datalink
                    List of installed extensions
   Name   | Version |  Schema  |             Description             
----------+---------+----------+-------------------------------------
 datalink | 0.3     | datalink | SQL/MED DATALINK type and functions
(1 row)

select p.pronamespace::regnamespace,p.oid::regprocedure,l.lanname,obj_description(p.oid) 
  from pg_proc p 
  join pg_language l on p.prolang=l.oid
 where pronamespace = 'datalink'::regnamespace
    or (pronamespace = 'pg_catalog'::regnamespace 
        and proname like 'dl%'
        and obj_description(p.oid) like 'SQL/MED%')
 order by 1, obj_description(p.oid) is null, cast(p.oid::regprocedure as text) collate "C";
 pronamespace |                                             oid                                              | lanname |                                                        obj_description                                                         
--------------+----------------------------------------------------------------------------------------------+---------+--------------------------------------------------------------------------------------------------------------------------------
 pg_catalog   | dlcomment(datalink)                                                                          | sql     | SQL/MED - Returns the comment value, if it exists, from a DATALINK value
 pg_catalog   | dlnewcopy(datalink,integer)                                                                  | plpgsql | SQL/MED - Returns a DATALINK value which has an attribute indicating that the referenced file has changed.
 pg_catalog   | dlpreviouscopy(datalink,integer)                                                             | plpgsql | SQL/MED - Returns a DATALINK value which has an attribute indicating that the previous version of the file should be restored.
 pg_catalog   | dlurlcomplete(datalink)                                                                      | sql     | SQL/MED - Returns the data location attribute (URL) from a DATALINK value
 pg_catalog   | dlurlcompleteonly(datalink)                                                                  | sql     | SQL/MED - Returns the data location attribute (URL) from a DATALINK value
 pg_catalog   | dlurlpath(datalink)                                                                          | sql     | SQL/MED - Returns the file path from a DATALINK value
 pg_catalog   | dlurlpathonly(datalink)                                                                      | sql     | SQL/MED - Returns the file path from a DATALINK value
 pg_catalog   | dlurlscheme(datalink)                                                                        | sql     | SQL/MED - Returns the scheme from a DATALINK value
 pg_catalog   | dlurlserver(datalink)                                                                        | sql     | SQL/MED - Returns the file server from a DATALINK value
 pg_catalog   | dlvalue(text,dl_linktype,text)                                                               | sql     | SQL/MED - Construct a DATALINK value
 datalink     | dl_chattr(name,name,name,dl_lco)                                                             | plpgsql | Set attributes for datalink column (buggy)
 datalink     | dl_lco(dl_link_control,dl_integrity,dl_read_access,dl_write_access,dl_recovery,dl_on_unlink) | sql     | Calculate dl_lco from individual options
 datalink     | dl_link_control_options(dl_lco)                                                              | sql     | Calculate dl_link_control_options from dl_lco
 datalink     | uri_get(text,text)                                                                           | plperlu | Get (extract) parts of URI
 datalink     | uri_set(text,text,text)                                                                      | plperlu | Set (replace) parts of URI
 datalink     | curl_get(text,boolean)                                                                       | plperlu | 
 datalink     | dl_class_adminable(regclass)                                                                 | sql     | 
 datalink     | dl_event_trigger()                                                                           | plpgsql | 
 datalink     | dl_newtoken()                                                                                | sql     | 
 datalink     | dl_ref(datalink,dl_lco,regclass,name)                                                        | plpgsql | 
 datalink     | dl_ri_trigger()                                                                              | plpgsql | 
 datalink     | dl_sql_advice()                                                                              | sql     | 
 datalink     | dl_unref(datalink,dl_lco,regclass,name)                                                      | plpgsql | 
(23 rows)

